### PR TITLE
[AutoImport] [Renaming] Skip remove used use statement on annotation during rename + auto import when no replacement on auto import

### PR DESCRIPTION
**Pull request by samsonasik**
Added test or testing skip remove use statement part of rename during auto import.

The rename annotation is allowed only on specific:

- symfony assert
- doctrine
- and serializer
  
see 

https://github.com/rectorphp/rector-src/blob/d55a35bcdede830d3927de1c11e0f7f0d12ee9e4/packages/BetterPhpDocParser/PhpDocManipulator/PhpDocClassRenamer.php#L36-L38
         
see 

- https://github.com/rectorphp/rector-symfony/issues/535#issuecomment-1762822651

on auto import, the process is overlapped, which annotation not renamed, but use statement removed during auto import.

### CODE REVIEW

**Comment by samsonasik** [comment]
@jambonfarci  @stefantalen  it fixed by skip remove use statement when no replacement during auto import https://github.com/rectorphp/rector-src/pull/5168/commits/cd3ee415c046e31f180f1d47b45cad2e919cf9f4 :tada:

File: rules/CodingStyle/Application/UseImportsRemover.php [hunk 1]
```diff
@@ -34,14 +44,33 @@ public function removeImportsFromStmts(array $stmts, array $removedUses): array
...
      * @param string[] $removedUses
+     * @param AliasedObjectType[]|FullyQualifiedObjectType[] $useImportTypes
      */
-    private function removeUseFromUse(array $removedUses, Use_ $use): Use_
+    private function removeUseFromUse(array $removedUses, Use_ $use, array $useImportTypes): Use_
     {
+        // nothing to remove, as no replacement
+        if ($useImportTypes === []) {
+            return $use;
+        }
```
Comment (samsonasik) [inline rules/CodingStyle/Application/UseImportsRemover.php:54]: on auto import enabled, when nothing to replace, the removing use statement should not happen, as it cause invalid removal on rename process.

File: packages/PostRector/Application/PostFileProcessor.php [hunk 2]
```diff
@@ -31,11 +31,11 @@ public function __construct(
...
         $this->postRectors = [
             // priority: 650
-            $classRenamingPostRector,
-            // priority: 600
             $nameImportingPostRector,
-            // priority: 500
+            // priority: 600
             $useAddingPostRector,
+            // priority: 500
+            $classRenamingPostRector,
```
Comment (samsonasik) [inline packages/PostRector/Application/PostFileProcessor.php:38]: the re-index to set name importing and use adding post rector first is on purpose to know that auto import collecting the statements that need to be added, which class renaming depends.

**Comment by samsonasik** [comment]
All checks have passed :tada:  @TomasVotruba I am merging it to have faster feedback to test ;)

**Comment by TomasVotruba** [comment]
:+1:

**Comment by samsonasik** [comment]
@TomasVotruba @jambonfarci @stefantalen I tested in laminas-servicemanager-migration usage on auto import usage, it seems cause error:

```diff
There were 5 failures:

1) LaminasTest\ServiceManager\Migration\Rector\RenameClassRector\AutoImportTest::test with data set #0 ('/Users/samsonasik/www/laminas...hp.inc')
Failed on fixture file "factory_invoke_typehint.php.inc"
Failed asserting that string matches format description.
--- Expected
+++ Actual
@@ @@
 
 namespace LaminasTest\ServiceManager\Migration\Rector\RenameClassRector\FixtureAutoImport;
 
-use Psr\Container\ContainerInterface;
+use Interop\Container\ContainerInterface;
 
 class FactoryInvokeTypehint
 {
-    public function __invoke(ContainerInterface $container)
+    public function __invoke(\Psr\Container\ContainerInterface $container)
```

which auto import should be allowed, this definitely need rework, I will check if the original allow change `SpacelessPhpDocTagNode` on PR:

- https://github.com/rectorphp/rector-src/pull/5164

can be solution, but on consequency, `rector-symfony` list need update for `AnnotationToAttribute` to use targetted:

```
Symfony\Component\Security\Http\Attribute\IsGranted
```

instead of 

```
Sensio\Bundle\FrameworkExtraBundle\Configuration\IsGranted
```

https://github.com/rectorphp/rector-symfony/blob/873458ca73a73f9becd1680963a347bbcb8e5f43/config/sets/sensiolabs/annotations-to-attributes.php#L14

Give me time :)

**Comment by samsonasik** [comment]
Reverting at https://github.com/rectorphp/rector-src/pull/5179

**Pull request merged**